server/migrations/community/c13819c566e7_migrate_away_from_jsonb_columns.py

```diff
@@ -0,0 +1,387 @@
+"""Migrate away from jsonb columns in project and project_version tables, introduce new file history table
+
+Revision ID: c13819c566e7
+Revises: 07f2185e2428
+Create Date: 2024-06-06 16:05:59.565541
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "c13819c566e7"
+down_revision = "07f2185e2428"
+branch_labels = None
+depends_on = None
+
+
+push_change_type = postgresql.ENUM(
+    "create", "update", "delete", "update_diff", name="push_change_type"
+)
+conn = op.get_bind()
+
+
+def upgrade():
+    op.create_table(
+        "file_history",
```

Comment:
It is 1:n with project version and author + date is store there (it would be unnecessary duplication if stored per file). Maybe the name of table is not great but we did not come up with anything better :-)

`mtime`  was dropped as it is not needed anymore, for the sake of clients we generate it from version datetime (it's been a fake mtime for quite some time)

`sanitized_path` was dropped in favor of `location` which it more important for backend - to know where to look for file, the sanitized path is however still kept as kind of hint in Upload table (before files are uploaded to server)